database-service/tests/utils/migrations.py

```diff
@@ -0,0 +1,24 @@
+import os
+import sh
+
+
+class HasuraClient:
+    def __init__(self, hasura_endpoint):
+        self.__cli = sh.Command('hasura')
+        self.__endpoint = hasura_endpoint
+
+    def apply_migrations(self, migrations_folder):
+        result = self.__cli('migrate', 'apply', '--endpoint', self.__endpoint,
+                            '--project', migrations_folder, '--skip-update-check')
+        return result.exit_code
+
+    def __get_number_of_migrations_in_folder(self, folder):
+        return int(len(os.listdir(folder)) / 2)
+
+    def rollback_migrations(self, project_folder):
+        migrations_folder = os.path.join(project_folder, 'migrations')
+        nb_migrations = self.__get_number_of_migrations_in_folder(
+            migrations_folder)
+        result = self.__cli('migrate', 'apply', '--down', nb_migrations, '--endpoint',
```

Comment:
if you select the fixtures project, then hasura will list you the structural and the fixtures migrations. It will tell you that the structural migrations are not present in the sources but present in the database and that the fixtures migrations are present both in the sources and the database. If you migrate down all, then you will migrate down the fixtures and the structural migrations. Because the structural migrations are not in project fixtures, when you migrate down all from fixtures project, you will have a failure, because the structural migrations cannot be found in the fixtures project, so hasura will not be able to migrate them down.